src/core/public/rendering/rendering_service.mock.ts

```diff
@@ -17,29 +17,25 @@
  * under the License.
  */
 
-import { chromeNavControlsRegistry } from '../../registry/chrome_nav_controls';
-import { uiModules } from '../../modules';
+import { RenderingStart, RenderingService } from './rendering_service';
 
-export function kbnAppendChromeNavControls() {
+const createStartContractMock = () => {
+  const setupContract: jest.Mocked<RenderingStart> = {
+    legacyTargetDomElement: document.createElement('div'),
+  };
+  return setupContract;
+};
 
-  uiModules
-    .get('kibana')
-    .directive('kbnChromeAppendNavControls', function (Private) {
-      return {
-        template: function ($element) {
-          const parts = [$element.html()];
-          const controls = Private(chromeNavControlsRegistry);
+type RenderingServiceContract = PublicMethodsOf<RenderingService>;
+const createMock = () => {
```

Comment:
This can be shortened to:

```ts
const createMock = (): jest.Mocked<RenderingServiceContract> => ({
  start: jest.fn().mockReturnValue(createStartContractMock()),
});
```

Reply:
Problem with the shortened way is it doesn't type check the value passed to `mockReturnValue` to make sure it satisfies the return type of `start`.